updatehub/src/settings.rs

```diff
@@ -64,8 +62,8 @@ impl Settings {
     /// Loads the settings from the filesystem. If
     /// `/etc/updatehub.conf` does not exists, it uses the default
     /// settings.
-    pub fn load() -> Result<Self> {
-        let path = std::path::Path::new(SYSTEM_SETTINGS_PATH);
+    pub fn load(path: std::path::PathBuf) -> Result<Self> {
```

Comment:
Please import  `PathBuf` so we simplify the import. Also, there is no need to shadow binding the `path` copying it to `Path`.